src/algebra/lie/free.lean

```diff
@@ -225,4 +220,19 @@ by { rw ← lift_symm_apply, exact (lift R).apply_symm_apply F, }
 have h' : (lift R).symm F₁ = (lift R).symm F₂, { ext, simp [h], },
 (lift R).symm.injective h'
 
+variables (R X)
+
+local notation `U` := universal_enveloping_algebra R (free_lie_algebra R X)
+local notation `A` := free_algebra R X
+
+/-- The universal enveloping algebra of the free Lie algebra is just the free unital associative
+algebra. -/
+@[simps] def universal_enveloping_equiv_free_algebra :
+  universal_enveloping_algebra R (free_lie_algebra R X) ≃ₐ[R] free_algebra R X :=
+alg_equiv.of_alg_hom
+  (universal_enveloping_algebra.lift R $ free_lie_algebra.lift R $ free_algebra.ι R   : U →ₐ A)
+  (free_algebra.lift R $ (universal_enveloping_algebra.ι R) ∘ (free_lie_algebra.of R) : A →ₐ U)
```

Comment:
I'm a little surprised these annotations are needed - does lean let you replace any of the `A` and `U`s with an underscore?

Reply:
Oh they aren't necessary!

I suspect I erroneously concluded they were necessary because I was building up the term using `#check` (where they are presumably necessary because `X` cannot be inferred) but in the `def` since the final type is specified the elaborator has enough information.

In any case, they're gone!